bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1034,19 +1033,39 @@ void PreProcessor::registerAndHandleClientPreProcessReqOnNonPrimary(ClientPrePro
   }
 }
 
-const char *PreProcessor::getPreProcessResultBuffer(uint16_t clientId,
-                                                    ReqId reqSeqNum,
-                                                    uint16_t reqOffsetInBatch) const {
-  // Pre-allocated buffers scheme:
+const char *PreProcessor::isAllocatedPreProcessResultBuffer(uint16_t clientId,
```

Comment:
No need for this function. You still have the same bufferOffset, the log, 
the "if (!preProcessResultBuffers_[bufferOffset].first)" in getPrePro.cessResultBuffer.

Reply:
Done.